### PR TITLE
InputRenderers

### LINKED ISSUE #1010: WNumberField should not expose size
Input in the number state do not implement the size attribute. See https://html.spec.whatwg.org/multipage/forms.html#input-type-attr-summary. This attribute and its accessors should be removed.

### PR DESCRIPTION
**Pull request by marksreeves**
1. Update renderers for fields which extend `Input` so that if the rendered field is in a read-only state only those attributes relevant to the state are rendered. Reolves #781.

2. Update WNumberField, its renderer and its schema to remove the output of a `size` attribute. The internal columns member and its accessors are deprecated. Fixes #1010.

@jonathanaustin PTAL

### CODE REVIEW

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WDropdownRenderer.java [hunk 1]
```diff
@@ -27,26 +27,29 @@
...
-		xml.appendOptionalAttribute("submitOnChange", dropdown.isSubmitOnChange(), "true");
-		xml.appendOptionalAttribute("tabindex", component.hasTabIndex(), component.getTabIndex());
-		xml.appendOptionalAttribute("toolTip", dropdown.getToolTip());
-		xml.appendOptionalAttribute("accessibleText", dropdown.getAccessibleText());
-		xml.appendOptionalAttribute("optionWidth", optionWidth > 0, optionWidth);
-		xml.appendOptionalAttribute("type", getTypeAsString(dropdown.getType()));
+		if (readOnly) {
+			xml.appendAttribute("readOnly", "true");
+		} else {
+			xml.appendOptionalAttribute("data", dataKey != null && !readOnly, dataKey);
```
Comment (jonathanaustin) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WDropdownRenderer.java]: The !readOnly can be removed now.

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WSingleSelectRenderer.java [hunk 2]
```diff
@@ -34,16 +34,19 @@ public void doRender(final WComponent component, final WebXmlRenderContext rende
...
-		xml.appendOptionalAttribute("readOnly", readOnly, "true");
-		xml.appendOptionalAttribute("submitOnChange", listBox.isSubmitOnChange(), "true");
-		xml.appendOptionalAttribute("tabIndex", component.hasTabIndex(), listBox.getTabIndex());
-		xml.appendOptionalAttribute("toolTip", component.getToolTip());
-		xml.appendOptionalAttribute("accessibleText", component.getAccessibleText());
-		xml.appendOptionalAttribute("rows", rows >= 2, rows);
+		if (readOnly) {
+			xml.appendAttribute("readOnly", "true");
+		} else {
+			xml.appendOptionalAttribute("data", dataKey != null && !readOnly, dataKey);
```
Comment (jonathanaustin) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WSingleSelectRenderer.java]: This !readOnly can also be removed now

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WMultiFileWidgetRenderer.java [hunk 3]
```diff
@@ -35,40 +35,45 @@ public void doRender(final WComponent component, final WebXmlRenderContext rende
...
+			if (dropzone != null) {
+				xml.appendAttribute("dropzone", dropzone.getId());
+			}
+			if (editor != null) {
+				xml.appendAttribute("editor", editor.getId());
+				if (editor.getUseCamera()) {
+					xml.appendAttribute("camera", true);
+				}
+			}
+			if (widget.getFileAjaxAction() != null) {
```
Comment (jonathanaustin) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WMultiFileWidgetRenderer.java]: We need to check if the AJAX should be honored in Readonly.

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WDropdownRenderer.java (hunk 1 above)
Comment (marksreeves) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WDropdownRenderer.java]: ok

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WSingleSelectRenderer.java (hunk 2 above)
Comment (marksreeves) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WSingleSelectRenderer.java]: ok

File: wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WMultiFileWidgetRenderer.java (hunk 3 above)
Comment (marksreeves) [inline wcomponents-core/src/main/java/com/github/bordertech/wcomponents/render/webxml/WMultiFileWidgetRenderer.java]: ok - this has exposed another bug see #1060

**Comment by jonathanaustin** [comment]
LGTM.

Thanks for making the changes.

**Pull request merged**